library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphTest.java

```diff
@@ -224,19 +396,122 @@ public void shouldAddAndGetEdge() {
         // Then
         final Edge edge = edges.next();
         assertFalse(edges.hasNext()); // there is only 1 vertex
-        assertEquals(VERTEX_1, ((List) edge.id()).get(0));
-        assertEquals(VERTEX_2, ((List) edge.id()).get(1));
-        assertEquals(CREATED_EDGE_GROUP, edge.label());
-        assertEquals(gafferPopInVertex, edge.inVertex());
-        assertEquals(gafferPopOutVertex, edge.outVertex());
-        assertEquals(1.5, (Double) edge.property(WEIGHT_PROPERTY).value(), 0);
+        assertThat(((List) edge.id()).get(0)).isEqualTo(VERTEX_1);
+        assertThat(((List) edge.id()).get(1)).isEqualTo(VERTEX_2);
+        assertThat(edge.label()).isEqualTo(CREATED_EDGE_GROUP);
+        assertThat(edge.inVertex()).isEqualTo(gafferPopInVertex);
+        assertThat(edge.outVertex()).isEqualTo(gafferPopOutVertex);
+        assertThat((Double) edge.property(WEIGHT_PROPERTY).value()).isEqualTo(1.5);
+    }
+
+    @Test
+    public void shouldGetEdgeInGroup() {
+        // Given
+        final Graph gafferGraph = getGafferGraph();
+        final GafferPopGraph graph = GafferPopGraph.open(TEST_CONFIGURATION_1, gafferGraph);
+        final GafferPopEdge edgeToAdd1 = new GafferPopEdge(CREATED_EDGE_GROUP, VERTEX_1, VERTEX_2, graph);
+        graph.addEdge(edgeToAdd1);
+
+        // When
+        final Iterator<GafferPopEdge> edges = graph.edges(VERTEX_1, Direction.OUT, CREATED_EDGE_GROUP);
+
+        // Then
+        final List<Edge> edgesList = new ArrayList<>();
```

Comment:
Might be nicer to instead do `assertThat(Lists.newArrayList(edges)).contains(edgeToAdd1)`

Reply:
Not really as that uses Guava. It would be better to just do `assertThat(edges).toIterable().contains(edgeToAdd1)`. I don't think it needs to be done in this PR though as there are a lot of places requiring this change. This class will be refactored as part of #3107 and these kinds of changes can be done in there.

Reply:
That does look better, I just don't like adding a while loop to make the iterable